### ISSUE TITLE
When including a role in a loop over a dictionary, content of the dictionary may make the play sequential

### ISSUE THREAD

**Issue by UnitedMarsupials-zz** [ansible/ansible#44229]
##### SUMMARY
When looping over a dictionary, if the entries are host-specific, the role may be applied to machines sequentially, rather than in parallel

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
include_role

##### ANSIBLE VERSION
```
ansible 2.6.2
  config file = /mi/A/ansible.cfg
  configured module search path = ['/mi/A/playbooks/library']
  ansible python module location = /opt/lib/python3.6/site-packages/ansible
  executable location = /opt/bin/ansible
  python version = 3.6.6 (default, Aug 14 2018, 14:36:33) [GCC 4.2.1 Compatible FreeBSD Clang 6.0.1 (tags/RELEASE_601/final 335540)]
```

##### CONFIGURATION
ALLOW_WORLD_READABLE_TMPFILES(/mi/A/ansible.cfg) = True
DEFAULT_FORKS(/mi/A/ansible.cfg) = 11
DEFAULT_MODULE_PATH(/mi/A/ansible.cfg) = ['/mi/A/playbooks/library']
DEFAULT_TIMEOUT(/mi/A/ansible.cfg) = 23

##### OS / ENVIRONMENT
N/A

##### STEPS TO REPRODUCE
Consider a task, which includes an auxiliary role in a loop over a dictionary, with several tasks applied during each iteration. Depending on the contents of the dictionary, Ansible may choose to apply the role to the inventory in sequence, rather than in parallel.

```yaml
- hosts: all
  gather_facts: false
  tasks:
  - set_fact:
      rabbits:
        uno:
          name: Flopsy
          prefers: "{{ inventory_hostname }}"
        dos:
          name: Mopsy
          prefers: "{{ inventory_hostname }}"
        tres:
          name: Cottontail
          prefers: "{{ inventory_hostname }}"

- hosts: all
  gather_facts: false
  tasks:
  - name: First round
    include_role:
      name: Lister
    with_dict: "{{ rabbits }}"
```
The `Lister`-role can be trivial:
```yml
- name: Task one
  shell: ":"

- name: Task two
  debug:
    msg: "Looping over {{ item.key }} on {{ inventory_hostname }}"
```

##### EXPECTED RESULTS
```
PLAY [all] **********************************************************************************************************************************************************

TASK [set_fact] *****************************************************************************************************************************************************
ok: [host1]
ok: [host2]

PLAY [all] **********************************************************************************************************************************************************

TASK [include_role : Lister] ****************************************************************************************************************************************

TASK [Lister : Task one] ********************************************************************************************************************************************
changed: [host2]
changed: [host1]

TASK [Lister : Task two] ********************************************************************************************************************************************
ok: [host1] => {
    "msg": "Looping over uno on host1"
}
ok: [host2] => {
    "msg": "Looping over uno on host2"
}

TASK [Lister : Task one] ********************************************************************************************************************************************
changed: [host2]
changed: [host1]

TASK [Lister : Task two] ********************************************************************************************************************************************
ok: [host1] => {
    "msg": "Looping over dos on host1"
}
ok: [host2] => {
    "msg": "Looping over dos on host2"
}

TASK [Lister : Task one] ********************************************************************************************************************************************
changed: [host2]
changed: [host1]

TASK [Lister : Task two] ********************************************************************************************************************************************
ok: [host1] => {
    "msg": "Looping over tres on host1"
}
ok: [host2] => {
    "msg": "Looping over tres on host2"
}

PLAY RECAP **********************************************************************************************************************************************************
host1                  : ok=7    changed=3    unreachable=0    failed=0   
host2                  : ok=7    changed=3    unreachable=0    failed=0   
```

##### ACTUAL RESULTS
```
PLAY [all] **********************************************************************************************************************************************************

TASK [set_fact] *****************************************************************************************************************************************************
ok: [host1]
ok: [host2]

PLAY [all] **********************************************************************************************************************************************************

TASK [include_role : Lister] ****************************************************************************************************************************************

TASK [Lister : Task one] ********************************************************************************************************************************************
changed: [host1]

TASK [Lister : Task two] ********************************************************************************************************************************************
ok: [host1] => {
    "msg": "Looping over uno on host1"
}

TASK [Lister : Task one] ********************************************************************************************************************************************
changed: [host1]

TASK [Lister : Task two] ********************************************************************************************************************************************
ok: [host1] => {
    "msg": "Looping over dos on host1"
}

TASK [Lister : Task one] ********************************************************************************************************************************************
changed: [host1]

TASK [Lister : Task two] ********************************************************************************************************************************************
ok: [host1] => {
    "msg": "Looping over tres on host1"
}

TASK [Lister : Task one] ********************************************************************************************************************************************
changed: [host2]

TASK [Lister : Task two] ********************************************************************************************************************************************
ok: [host2] => {
    "msg": "Looping over uno on host2"
}

TASK [Lister : Task one] ********************************************************************************************************************************************
changed: [host2]

TASK [Lister : Task two] ********************************************************************************************************************************************
ok: [host2] => {
    "msg": "Looping over dos on host2"
}

TASK [Lister : Task one] ********************************************************************************************************************************************
changed: [host2]

TASK [Lister : Task two] ********************************************************************************************************************************************
ok: [host2] => {
    "msg": "Looping over tres on host2"
}

PLAY RECAP **********************************************************************************************************************************************************
host1                  : ok=7    changed=3    unreachable=0    failed=0   
host2                  : ok=7    changed=3    unreachable=0    failed=0   
```
Removing the `prefers`-entries from the `rabbits`-dictionary in this example is the way I got the "Expected" results listed. The dictionary can have multiple fields and the things will continue working correctly _unless_ somewhere the `inventory_hostname` is referenced again.

I don't see, why it should have such an effect.

**Comment by sivel** [comment]
Duplicate of https://github.com/ansible/ansible/issues/36978
